frontend/src/components/SystemSubmitDrawer/index.tsx

```diff
@@ -111,6 +115,34 @@ const filterFunc: FilterFunc<LanguageCode> = (query, data) => {
   return result;
 };
 
+/**
+ * Parse the form into fields so that client code generator can recognize
+ */
+function parseFormFields(
```

Comment:
It looks like `CodeGenFields` is almost the same as `FormData` except that it has an extra `use_custom_dataset` field. Could we pass `FormData` to `<ClientCodeDisplay />` and do the parsing over there (if necessary) instead?